tests/generative/test_query_model.py

```diff
@@ -252,13 +252,18 @@ def run_error_test(query_engines, data, variable):
         re.compile(".+Internal error: An expression services limit has been reached.+"),
     ),
     # Trino also raises an error if the variable strategy generates queries
-    # that result in many nested queries; again the many-date stacking seems to be the main culprit
+    # that result in many nested or too-long queries; again the many-date stacking seems to be the
+    # main culprit
     (
         sqlalchemy.exc.DBAPIError,
         re.compile(
             ".+TrinoQueryError.+the query may have too many or too complex expressions.+"
         ),
     ),
+    (
+        sqlalchemy.exc.ProgrammingError,
+        re.compile(".+TrinoUserError.+QUERY_TEXT_TOO_LARGE.+"),
```

Comment:
I wonder how much this is to with the fact the we're using CTEs as our "reify query" mechanism (as that's the default in BaseSQL). It means we end up with one enormous query at the end, rather than a series of smaller queries creating temporary tables.

I think this is the right thing to do for now. There's going to be a whole piece of work in figuring out how all this performs in production (which is likely to be _very_ different from how the MSSQL database performs) and optimising it. But there's no point spending time on that until we've got something solid to work with.